src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
+      victim.health -= 50;
+
+      if (victim.health <= 0) {
+        const index = Animal.alive.indexOf(victim);
+
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
I recommend improving the method of removing dead animals from the massif.
let's put this in a separate static method, changing this logic to remove dead animals from the Animal.alive array using filtering:

```
static removeDeadAnimals() {
    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
}
```
This method is called inside the `bite` method:
```suggestion
      Animal.removeDeadAnimals();
```